pgml-docs/docs/blog/scaling-postgresml-to-one-million-requests-per-second.md

```diff
@@ -0,0 +1,252 @@
+);
+```
+
+and fetched our predictions the usual way
+
+```postgresql
+SELECT pgml.predict(
+	'flights',
+	ARRAY[
+		year,
+		quarter,
+		month,
+		distance,
+		dayofweek,
+		dayofmonth,
+		flight_number_operating_airline,
+		originairportid,
+		destairportid,
+		flight_number_marketing_airline,
+		departure
+	]
+) AS prediction
+FROM flights_mat_3 LIMIT :limit;
+```
+
+where `:limit` is the batch size of 1, 5, and 20, depending on the benchmark.
+
+#### Model
+
+The model is roughly the same as the one we used in our previous [post](/blog/postgresml-is-8x-faster-than-python-http-microservices), with just one extra feature added, which improved R<sup>2</sup> a little bit.
+
+### Hardware
+
+#### Client
+The client was a `c5n.4xlarge` box on EC2. We chose the `c5n` class to have the 100 GBit NIC, since we wanted it to saturate our network as much as possible. Thousands of clients were simulated using [`pgbench`](https://www.postgresql.org/docs/current/pgbench.html) with `-c n` where `n` is the number of clients.
+
+#### PgCat Pooler
+PgCat, written in asynchronous Rust, was running on `c5.xlarge` machines (4 vCPUs, 8GB RAM) with 4 Tokio workers. We used between 1 and 35 machines, and scaled them in increments of 5-20 at a time.
+
+The pooler did a decent amount of work around parsing queries, making sure they are read-only `SELECT`s, and routing them, at random, to replicas. If any replica was down for any reason, it would route around it to remaining machines.
+
+#### Postgres Replicas
+Postgres replicas were running on `c5.9xlarge` machines with 36 vCPUs and 72 GB of RAM. The hot dataset fits entirely in memory. The servers were intentionally saturated to maximum capacity before scaling up to test queuing and graceful degradation of performance.
+
+
+#### Raw Results
+
+Raw latency data is available [here](https://static.postgresml.org/benchmarks/reads-latency.csv) and raw throughput data is available [here](https://static.postgresml.org/benchmarks/reads-throughput.csv).
+
+## Feedback
+
+Many thanks and ❤️ to all those who are supporting this endeavor. We’d love to hear feedback from the broader ML and Engineering community about applications and other real world scenarios to help prioritize our work. You can show your support by starring us on our [Github](https://github.com/postgresml/postgresml/).
+
+
+## We're Hiring!
+
+[PostgresML](https://github.com/postgresml/postgresml/) and [PgCat](https://github.com/levkk/pgcat/) are free and open source, and to support their development, and many more things we're building, we started a company. We're only a few months old, and we have raised enough funding to say, for the first time ever: we're hiring!
+
+We're looking for software engineers interested in machine learning, solving big problems, databases, and anything in between. Don't hesitate to reach out to <a href="mailto:team@postgresml.org">team@postgresml.org</a> or in [Discord](https://discord.gg/DmyJP3qJ7U).
+
+## Call to Early Adopters
```

Comment:
I think this is the best CTA, I'd ## Feedback into this one, and drop the We're Hiring!